pa4/checker-pa4.ml

```diff
@@ -286,10 +307,21 @@ let add_method_types meth_env input_class = begin
 	List.iter(fun own_feat -> 
 		match own_feat with
 		|  Method ((_, m_name), formals, (_, return_type),_) -> 
-			let formal_types = List.map (fun (_, (_, fname)) -> Class fname) formals in
-			let key: string * string = (class_name, m_name) in
-			let value: static_type list= formal_types  @  [Class return_type] in
-			Hashtbl.add meth_env key value
+			let formal_types = List.map (fun (_,(_,fname)) -> 
+					if fname = class_name then 
+						SELF_TYPE fname
+					else 
+						Class fname
+				) formals in
+			let key = (class_name,m_name) in
```

Comment:
Since we don't reuse this value I would refer to see it defined as an expression on line 324 rather than as a variable.